src/redux/selectors.js

```
@@ -88,4 +91,31 @@ export const userHasVerifiedSourceOfFunds: Selector<
   },
 );
 
+function getAccountBalances(accounts: Array<AccountType>, currency: string) {
+  return accounts.map((account: AccountType) =>
+    new Account(account).getRepresentationalBalance(currency),
+  );
+}
+
+function getAccountBalance(
+  accounts: Array<AccountType>,
+  currency: string,
+): number {
+  const accountBalances = getAccountBalances(accounts, currency);
+  return Number(
+    accountBalances
+      .reduce((balance1, balance2) => balance1 + balance2, 0)
+      .toFixed(2),
+  );
+}
+
+export const getTotalAccountBalance: Selector<*, *, number> = createSelector(
+  [getAccounts, getAccountsCurrency],
+  (accounts: Array<AccountType>, currency: string): number => {
+    if (accounts.length) {
+      return getAccountBalance(accounts, currency);
+    }
+    return -1;
```

Maybe throwing an Error would be better here? _Fail fast_ instead of showing a negative balance to the user (and without reporting it to sentry).